src/gridui.h

```diff
@@ -153,6 +132,86 @@ class _GridUi {
     uint16_t m_tab;
 };
 
+template <>
+inline builder::Switcher* _GridUi::newWidget<builder::Switcher>(float x, float y, float w, float h, uint16_t uuid, uint16_t tab) {
+    using T = builder::Switcher;
+    static_assert(std::is_base_of<builder::Widget, T>::value, "T must inherit from builder::Widget.");
+
+    std::lock_guard<std::mutex> l(m_states_mu);
+
+    if (!checkUuidFreeLocked(uuid))
+        uuid = generateUuidLocked();
+
+    auto* state = new WidgetState(uuid, x, y, w, h, tab, &T::callbackTrampoline);
+    m_states.push_back(std::unique_ptr<WidgetState>(state));
+
+    auto* widget = new T(T::name(), *state);
+    m_widgets.push_back(std::unique_ptr<T>(widget));
+
+    if (std::is_same<T, builder::Switcher>::value) {
+        widget->addCallback("changed", [&](gridui::Switcher& w) {
+            changeTab(w.value());
+        });
+    }
+    return widget;
+}
```

Comment:
Pokud vidím dobře, tak je přidané jen handlování toho callbacku, co to nechat jak to je všechno původně, ale udělat tohle v builder metodě:

```cpp
builder::Switcher& switcher(float x, float y, float w, float h, uint16_t uuid = 0, uint16_t tab = 0) {
    auto *switcher = newWidget<builder::Switcher>(x, y, w, h, uuid, tab);
    switcher->addCallback("changed", [&](gridui::Switcher& w) {
            changeTab(w.value());
    });
    return *switcher;
}
```

Reply:
Jo, je pravda, že mi nedošlo, co tohle vlstně vrací...